sling/nlp/document/document.cc

```diff
@@ -342,6 +352,22 @@ void Document::RemoveMention(Handle handle, Span *span) {
   }
 }
 
+int Document::Locate(int position) const {
+  int index = 0;
+  int len = tokens_.size();
+  while (len > 0) {
+    int width = len / 2;
+    int mid = index + width;
+    if (tokens_[mid].begin() < position) {
```

Comment:
Since we have half-open token and span intervals, 1 is not really out of bounds in this case. I have implemented it this was to make it simple to use in the common case where you have then begin and (exclusive) end byte positions. In that case you can just say:
```
int span_begin = document->Locate(text_begin);
int span_end = document->Locate(text_end);
```
With this semantics, we don't need separate methods for the begin and end methods. This is one of the few simplifications you get from having half-open intervals.